File: frontend/fix_layout.py
import re

with open('src/components/Layout.tsx', 'r') as f:
    content = f.read()

# We know the first return block starts at line 85: "  return (\n    <div className=\"min-h-screen bg-background\">"
# and ends at line 245: "    </div>\n  )\n}"
# The second return block is from 248: "  return (\n    <div className=\"min-h-screen bg-background font-sans text-slate-900\">"
# to 318: "    </div>\n  )\n}"

pattern = r"  return \(\n    <div className=\"min-h-screen bg-background\">.*?(?=  return \(\n    <div className=\"min-h-screen bg-background font-sans text-slate-900\">)"

# let's just do a specific replace
lines = content.split('\n')
start1 = -1
end1 = -1
start2 = -1

for i, line in enumerate(lines):
    if 'return (' in line and 'bg-background' in lines[i+1] and 'font-sans' not in lines[i+1]:
        start1 = i
    if 'return (' in line and 'font-sans text-slate-900' in lines[i+1]:
        start2 = i
        
for i in range(start1, start2):
    if lines[i] == '}':
        end1 = i

if start1 != -1 and start2 != -1 and end1 != -1:
    new_lines = lines[:start1] + lines[start2:]
    with open('src/components/Layout.tsx', 'w') as f:
        f.write('\n'.join(new_lines))
    print("Fixed layout")
else:
    print(f"Failed to find indices: {start1}, {end1}, {start2}")

File: frontend/fix_layout.js
const fs = require('fs');
const content = fs.readFileSync('src/components/Layout.tsx', 'utf-8');
const lines = content.split('\n');

const newReturnStart = lines.findIndex(line => line.includes('  return (') && line.includes('min-h-screen bg-background font-sans'));
const firstReturnStart = lines.findIndex(line => line.includes('  return (') && line.includes('min-h-screen bg-background'));

if (firstReturnStart > -1 && newReturnStart > -1) {
    // wait, we just want to remove the old return block from 85 to 246
    // find the line of the first return
    const returnIndex = lines.findIndex(l => l.trim() === 'return (');
    const funcEndIndex = lines.findIndex((l, i) => i > returnIndex && l === '}');
    
    // Actually let's just do it manually with sed or similar, wait, javascript is easier.
}
